examples/rpc/partner-plugin/handlers/accommodation/v2/accommodation_search.go

```diff
@@ -0,0 +1,246 @@
+
+		// loop filtered properties and check if they are already in available_properties
+		for _, prop := range filtered_props {
+			// Check if property already exists in available_properties
+			exists := false
+			for _, existingProp := range available_properties {
+				if existingProp.Property.SupplierCode.SupplierCode == prop.Property.SupplierCode.SupplierCode {
+					exists = true
+					break
+				}
+			}
+			if !exists {
+				available_properties = append(available_properties, prop)
+			}
+		}
+
+		var price = 500
+
+		// generate search result
+		for _, prop := range available_properties {
+
+			// units requested
+			units_requested := query.UnitCount
+
+			// empty units array
+			units := make([]*accommodationv2.Unit, 0)
+
+			// loop all rooms
+			for _, room := range prop.Rooms {
+
+				units = append(units, &accommodationv2.Unit{
+					Type:             0,
+					SupplierRoomCode: room.SupplierCode,
+					SupplierRoomName: room.SupplierName,
+					OriginalRoomName: room.OriginalName,
+					TravelPeriod: &typesv1.TravelPeriod{
+						StartDate: &typesv1.Date{
+							Year:  query.TravelPeriod.GetStartDate().GetYear(),
+							Month: query.TravelPeriod.GetStartDate().GetMonth(),
+							Day:   query.TravelPeriod.GetStartDate().GetDay(),
+						},
+						EndDate: &typesv1.Date{
+							Year:  query.TravelPeriod.GetEndDate().GetYear(),
+							Month: query.TravelPeriod.GetEndDate().GetMonth(),
+							Day:   query.TravelPeriod.GetEndDate().GetDay(),
+						},
+					},
+					TravellerIds: getTravellerIds(query.Travellers),
+					Beds:         room.Beds,
+					PriceDetail: &typesv2.PriceDetail{
+						Price: &typesv2.Price{
+							Value: fmt.Sprintf("%d", price),
+							Currency: &typesv2.Currency{
+								Currency: req.SearchParametersGeneric.Currency.Currency,
+							},
+						},
+					},
+					Services:       []*typesv2.ServiceFact{},
+					MealPlanCode:   &typesv1.MealPlan{},
+					RatePlan:       &typesv1.RatePlan{},
+					RateRule:       &typesv1.RateRule{},
+					CancelPolicies: []*typesv2.CancelPolicy{},
+					RemainingUnits: 0,
+					PropertyCode:   &typesv2.ProductCode{},
+					SupplierCode:   prop.Property.SupplierCode,
+					Remarks:        "",
+				})
+
+				price += 250
+
+				if units_requested == int32(len(units)) {
+					break
+				}
+			}
+
+			// check how many units are requested
+			if units_requested == int32(len(units)) {
+				searchResults = append(searchResults, &accommodationv2.AccommodationSearchResult{
+					ResultId: int32(len(searchResults) + 1),
+					QueryId:  query.QueryId,
+					TotalPriceDetail: &typesv2.PriceDetail{
+						Price: &typesv2.Price{
+							Value: fmt.Sprintf("%d", price),
+							Currency: &typesv2.Currency{
+								Currency: req.SearchParametersGeneric.Currency.Currency,
+							},
+						},
+					},
+					Units: units,
+				})
+			}
+		}
+	}
+
+	// generate a random string of 8 numbers
+	md.RequestID = fmt.Sprintf("%08d", rand.Intn(100000000))
```

Comment:
Altering the metadata `RequestID` (to match the [SearchResponseMetadata](https://github.com/chain4travel/camino-messenger-protocol/blob/c4t/proto/cmp/types/v2/search.proto#L82C16-L82C20 )) seems to cause a problem forwarding the response on distributor side when testing the full flow 

postman -> distributor -> supplier -> supplier-plugin
postman <-**X**- distributor <- supplier <-supplier-plugin

```
2024-11-12T13:22:53.149+0100	DEBUG	messaging/processor.go:122	Processing msg event of type: cmp.services.accommodation.v2.AccommodationSearchService.Response
2024-11-12T13:22:53.149+0100	DEBUG	messaging/processor.go:360	Forwarding outbound response message: 11989351
2024-11-12T13:22:53.149+0100	WARN	messaging/processor.go:369	Failed to forward message: no response channel for request (11989351)```